docs/modules/ROOT/pages/api/access.adoc

```diff
@@ -0,0 +1,487 @@
+:github-icon: pass:[<svg class="icon"><use href="#github-icon"/></svg>]
+:AccessControl: xref:AccessControl[AccessControl]
+:Ownable: xref:Ownable[Ownable]
+
+= Access Control
+
+[.readme-notice]
+NOTE: This document is better viewed at https://docs.openzeppelin.com/contracts/api/access
+
+This directory provides ways to restrict who can access the functions of a contract or when they can do it.
+
+- {AccessControl} provides a general role based access control mechanism. Multiple hierarchical roles can be created and
+assigned each to multiple accounts.
+- {Ownable} is a simpler mechanism with a single owner "role" that can be assigned to a single account.
+This simpler mechanism can be useful for quick tests but projects with production concerns are likely to outgrow it.
+
+== Authorization
+
+[.contract]
+[[Ownable]]
+=== `++Ownable++` link:https://github.com/OpenZeppelin/cairo-contracts/blob/cairo-2/src/access/ownable/ownable.cairo[{github-icon},role=heading-link]
+
+[.hljs-theme-light.nopadding]
+```javascript
+use openzeppelin::access::ownable::Ownable;
+```
+
+Contract module which provides a basic access control mechanism, where
+there is an account (an owner) that can be granted exclusive access to
+specific functions.
+
+This module provides an internal utility named `assert_only_owner`, which can be applied to your functions
+to restrict their use to the owner.
+
+[.contract-index]
+.Utilities
+--
+* xref:Ownable-assert_only_owner[`++InternalImpl::assert_only_owner(self)++`]
+--
+
+[.contract-index]
+.Functions
+--
+[.contract-subindex-inherited]
+.InternalImpl
+
+* xref:Ownable-initializer[`++initializer(self, owner)++`]
+* xref:Ownable-_transfer_ownership[`++_transfer_ownership(self, new_owner)++`]
```

Comment:
I feel like we should list the `InternalImpl` at the end of the `Functions` list. That's how they're listed in the accompanying account docs and the Solidity docs. Wdyt?

Reply:
In the previous iteration, I think we can keep the initializer at the top of `InternalImpl`...but if we're also throwing the constructor in there, something like this seems fitting:

```
constructor()                                         INTERNALIMPL
initializer()
_internal_func()
...
```

> If we follow the proposal of splitting External Functions and Internal Functions this will be solved as well.

And yeah, if we follow the proposal^, I think the approach you have works as well

Reply:
> also i don't see the point of having Utilities as a separate section, why not part of Internals?

My thinking is because those will be used more frequently than other internals, and is better to catch the user attention to those first IMO. I'm ok with moving them to internal if you think is not worth.

Reply:
> in the meantime maybe we can

In the UI, what you are suggesting is to have a section like Utilities just for the constructor? Because that's the only way i can think of for presenting it first, and it seems weird to me to have a section Constructor, with the constructor method alone.

Reply:
> > also i don't see the point of having Utilities as a separate section, why not part of Internals?
> 
> My thinking is because those will be used more frequently than other internals, and is better to catch the user's attention to those first IMO. I'm ok with moving them to internal if you think is not worth it.

i guess we can call them "asserts" or "guards", like modifiers. but anything can be a utility, like `_mint`

Reply:
Your call on using Asserts or removing the section, let me know to update accordingly.

Reply:
> In the UI, what you are suggesting is to have a section like Utilities just for the constructor? Because that's the only way i can think of for presenting it first, and it seems weird to me to have a section Constructor, with the constructor method alone.

Let me know what are you suggesting about constructors.

Reply:
> In the UI, what you are suggesting is to have a section like Utilities just for the constructor?

sorry, missed it. yes that's what i was suggesting

Reply:
> Your call on using Asserts or removing the section, let me know to update accordingly.

i'd remove it for consistency with the rest of the sections. we can [add guards in a future iteration](https://github.com/OpenZeppelin/cairo-contracts/issues/745).